up-l1/mqtt_5.adoc

```diff
@@ -0,0 +1,135 @@
+| UserProperty
+| ("10", "{UAttribute.token}") # Optional token
+| UserProperty
+| ("11", "{UAttribute.traceParent}") # Optional traceparent
+|===
+
+All uAttributes are mapped to a MQTT header UserProperty, where the key is the UAttribute protobuf field number. The value is a string representation of the UAttribute field.
+
+=== URI Mapping to MQTT topics
+
+The MQTT topic a message is published on utilizes elements from both the source and sink UUri fields. Which elements from the UUri are used in the topic depend on whether the sink UUri is present. In addition, there is a client identifier that defines where the uTransport client is located.
+
+Has sink UUri:
+
+`{client_identifier}/{source.authority_name}/{sink.authority_name}/{sink.ue_id}/{sink.ue_version_major}/{sink.resource_id}`
+
+No sink UUri (Publish message type only):
+
+`{client_identifier}/{source.authority_name}//{source.ue_id}/{source.ue_version_major}/{source.resource_id}`
+
+The entity and resource elements always represent the destination of the message. In the case of a Publish message type, the destination happens to be based on the source UUri. The source authority information is also always present to allow for authorization controls based on the authority information, since an authority maps to the uE sending the message.
+
+The client_identifier is used to determine where messages are coming from. The valid values for the client_identifier are:
+
+- `c` : Indicates that the message originated from the cloud
+- `d` : Indicates that the message originated from a device and is remote
+- `l` : Indicates that the message originated from a device and is local to that device
+
+The client_identifier that is used for the topic is based on the kind of MQTT uTransport client that is sending the message. An MQTT uTransport client can be one of the following:
+
+- Cloud MQTT uTransport client: A client that runs in the cloud.
+- Remote MQTT uTransport client: A client that runs on a device and is connected to a cloud MQTT broker.
+- Local MQTT uTransport client: A client that runs on a device and is connected to a local MQTT broker.
+
+Below are examples of messages being sent and how to subscribe to those messages:
+
+==== Remote uTransport Client Examples
+
+How a source and sink UUri are mapped to an MQTT topic when the message is sent from a remote uTransport client:
+
+[cols="1,1,1,1"]
+|===
+| source | sink | send topic | listen topic
+| /vehicle_1.veh.com/1234/1/5678 | /vehicle_2.veh.com/4321/1/8765 | d/vehicle_1.veh.com/vehicle_2.veh.com/4321/1/8765 | c/vehicle_1.veh.com/vehicle_2.veh.com/4321/1/8765
+| /vehicle_1.veh.com/1234/1/5678 | None | d/vehicle_1.veh.com//1234/1/5678 | c/vehicle_1.veh.com//1234/1/5678/
+|===
+
+Listener examples with wildcards:
+
+[cols="1,1,1,1"]
+|===
+| goal | source filter | sink filter | listener topic
+| Subscribe to all incoming message to a UAuthority | empty | /vehicle_1.veh.com/FFFFFFFF/FF/FFFF | c/\+/vehicle_1.veh.com/+/\+/+
```

Comment:
The sink filter needs to be
`//vehicle_1.veh.com/FFFF/FF/FFFF`
in this case. The service instance ID wildcard is 0x0000.